pkg/controller/manifestwork/manifestwork_controller.go

```diff
@@ -81,7 +83,17 @@ func (r *ReconcileManifestWork) Reconcile(ctx context.Context, request reconcile
 	}
 
 	if !managedCluster.DeletionTimestamp.IsZero() {
-		// the managed cluster is deleting, delete its manifestworks
+		// the managed cluster is deleting, delete its manifestWorks.
+		// all managedClusterAddons should be deleted before we delete the klusterlet ManifestWork since managedClusterAddons depends on ManifestWork.
+		addons := &addonv1alpha1.ManagedClusterAddOnList{}
```

Comment:
because we will stop the delete procession if there are addon manifestworks, my understanding only the addons is deleted, the manifestworks of addons will be deleted, so do we need this?

Reply:
or if we have this, could we remove the limitation for addon manifestworks? I think determine manifestworks may be  a better way?